model/index.js

```diff
@@ -91,25 +90,26 @@ module.exports = class Model extends EventEmitter {
       callback = body;
       body = undefined;
     }
-
     let settings = Object.assign({}, originalSettings);
     settings.timeout = settings.timeout || this.options.timeout;
-    settings.uri = settings.uri || settings.url || url.format(settings);
-    settings.body = settings.body || body || settings.data;
-
-    settings = _.omit(settings, urlKeys, 'data', 'url');
+    settings.url = settings.uri || settings.url || url.format(settings);
```

Comment:
Make setting key name configurable from properties file. To support backward compatibility.

Reply:
done